frontend/hub/approvals/hooks/useApproveCollections.tsx

```diff
@@ -67,15 +82,32 @@ export function approveCollection(
     )) as PulpItemsResponse<Repository>;
     approvedRepo = repoRes.results[0].pulp_href;
 
-    await hubAPIPost(
-      pulpAPI`/repositories/ansible/ansible/${
-        parsePulpIDFromURL(collection.repository?.pulp_href || '') || ''
-      }/move_collection_version/`,
-      {
-        collection_versions: [`${collection.collection_version?.pulp_href}`],
-        destination_repositories: [approvedRepo],
+    const pipeline = collection.repository?.pulp_labels?.pipeline;
+    if (pipeline == 'approved') {
+      throw new Error(t('You can only approve collections in rejected or staging repositories'));
+    }
+
+    if (repoRes.count > 1) {
+      if (bulkAction) {
+        throw new Error(
+          t(
+            'You can use bulk action only when there is single approved repo, but you have multiple approved repositories.'
+          )
+        );
+      } else {
+        copyToRepository(collection, 'approve');
       }
-    );
+    } else {
+      await hubAPIPost(
+        pulpAPI`/repositories/ansible/ansible/${
+          parsePulpIDFromURL(collection.repository?.pulp_href || '') || ''
```

Comment:
these seem rather more dangerous with post...

```suggestion
          parsePulpIDFromURL(collection.repository.pulp_href)
```

and make sure to have a condition that checks `collection.repository` first?